### ISSUE TITLE
How to distinguish between drag and click events

### ISSUE THREAD

**Issue by 12343954** [ducksboard/gridster.js#112]
I want the mouse down to 1 second or 2, and then began to drag, in order to distinguish the click event . or any other good ideas ?

**Comment by nichthinkof30** [comment]
hi,same problem I'm facing this issues too. Anyone can help on this??
here is my stackoverflow question.
http://stackoverflow.com/questions/14301026/gridster-do-not-invoke-the-click-action-for-the-div/14304615#comment19872451_14304615

**Comment by hoyer-io** [comment]
I fixed it with a trigger variable: var dragged = 0
Add the following to your gridster initialization which sets the variable to 1 on drag start:

...
    draggable: {
        start: function(event, ui) {  
            dragged = 1;
            // DO SEOMETHING
         }
    }
....

In your click event on the same object, check:
...
    if(!dragged){
        // DO SOMETHING
    }
    // RESET DRAGGED SINCE CLICK EVENT IS FIRED AFTER drag stop
    dragged = 0
....

**Comment by 12343954** [comment]
Unfortunately, @colorfool-me , your method does not work .
May be @ducksboard  did not take into account this problem .
But the dialog plugin of jquery does not have this problem，I have been tested
I looked dialog plug-in source code, inside depends on jquery.mouse plug
maybe jquery improve the mouse event
However, even if this problem is fixed, but on the touch device, but also have this problem . @nichthinkof30

**Comment by hoyer-io** [comment]
Sorry I missed the point with the seconds. My method is just to distinguish click & drag events.
Just an idea: 
You could try to add a timer to your click event and if the time period is up, fire the drag event on your own?

**Comment by 12343954** [comment]
oh , thank you very much , bro ,@colorfool-me
I also do not want to distinguish between drag and click events with timestamp . 
I expect that there are better ways to solve this problem.

**Comment by 12343954** [comment]
oh . sorry @colorfool-me , it's my fault , i write the wrong code , you method is worked
Although , it is not the best way, but it solved my problem on PC browsers .
but on ipad , hers is no click effect , i dont know why, my code is ,

```
   var gridster ,tuozhuai ;
   $(function () {
        gridster = $(".gridster > ul").find("> li").click(function(e){
            !tuozhuai && alert(1) //ipad2 ，not show alert
            tuozhuai=0;
            }).end()
            .gridster({widget_margins: [10, 10],
            widget_base_dimensions: [140, 140],min_cols: 10,min_rows: 20
            ,serialize_params: function($w, wgd) {
                return {
                    id: wgd.el[0].id
                    //,col: wgd.col
                    //,row: wgd.row
                };
            }
            ,draggable: {
                start:function(event, ui){ tuozhuai=1; }
            }
        }).data('gridster');
```

**Comment by nichthinkof30** [comment]
but it still invoke the click after I stop dragging.

**Comment by hoyer-io** [comment]
Yeah @nichthinkof30 the event is fired, but that`s what the "if(!dragged){}" is for. Did you add this check?

**Comment by nichthinkof30** [comment]
im not sure im doing it correctly or not.

here is the piece of code.

  var gridster ,draggable ;
   $(function () {
        gridster = $(".gridster > ul").find("> li ").click(function(e){
            !draggable && alert(1) //ipad2 ，not show alert
            draggable=0;
            alert(draggable);
            }).end()
            .gridster({widget_margins: [5, 5],
            widget_base_dimensions: [128, 130],min_cols: 10,min_rows: 20
            ,serialize_params: function($w, wgd) {
                return {
                    id: wgd.el[0].id
                    //,col: wgd.col
                    //,row: wgd.row
                };
            }
            ,draggable: {
                start:function(event, ui){ 
                //  alert("hekio);");
                    draggable=1; }
            }
        }).data('gridster');
    //gridster.disable();
//   gridster.resize_widget($('.gridster li').eq(0), 1,1);
         if(!dragged){
            $("a#blue.grid").click(function(){
            window.location = '../yokotachi/category_list.php?category=yokosmart';
        });
            }
    // RESET DRAGGED SINCE CLICK EVENT IS FIRED AFTER drag stop
            dragged = 0
    });

**Comment by hoyer-io** [comment]
I think it should look like this:
$("a#blue.grid").click(function(){
  if(!dragged){
    window.location = '../yokotachi/category_list.php?category=yokosmart';
  }
  dragged = 0
});

**Comment by marcovanest** [comment]
if you want to disable the on_start_drag click try this (quick and dirty but it should work)

```
  var gridster = $(".gridster>ul").gridster().data('gridster');
  var olddragstart = gridster.on_start_drag;
  gridster.on_start_drag = function(e) 
  {

    this.$player.find('a').on('click', function(e){
      e.preventDefault();
    });

    var ret = olddragstart.apply(this, arguments);
    return ret;
  };
```

**Comment by herebebeasties** [comment]
I've updated the stack overflow answer with a solution to this issue: http://stackoverflow.com/a/25431791/217893
Complete with a working example: http://jsfiddle.net/g22caks6/15/

In my opinion this behaviour should be baked into gridster. I'll see if I can generate a pull request for this.

**Comment by amarkitanis** [comment]
How can this be acvhieved in the angular version?
